### ISSUE TITLE
Get Github Action functional (prev: Delete docs from development)

### ISSUE THREAD

**Issue by cgettings** [nycehs/ehs-data-portal-frontend-temp#211]
We've set up the site preview to use the `gh-pages` branch, with an automatic build on merge into `development` using GitHub Actions. Now we can delete docs from dev, and save ourselves the headache of merge conflicts caused by docs.

We can also maybe just delete the current `main`, then rename `development` to `main`.

**Comment by mmontesanonyc** [comment]
OK so per chat (and bringing things here to store):
- `gh-pages` serves the site
- Action successfully deploys to gh-pages upon merges to `development`
- `devcopy` no longer needed

So we can delete `docs` (and add it to `gitignore`) to tidy things up.

**Comment by mmontesanonyc** [comment]
Nota bene:

**Comment by mmontesanonyc** [comment]
OK, re-opening this. in short - I did the above steps and for some reason, the action that triggers a build/deployment to `gh-pages` isn't working properly. 

It runs, but the deployment step fails:
![image](https://user-images.githubusercontent.com/55593359/166259521-83fde19e-7d42-4f98-9c90-fc008cf96a81.png)

I tried tweaking a lot of things to see if it would work, but I think it might not be something I changed - I think when you got it to run on Friday, it still wasn't deploying:
![image](https://user-images.githubusercontent.com/55593359/166259659-75c51862-36fc-4f1e-ad1c-583868ca4b41.png)

But _something_ was deploying - stuff got to `gh-pages`:
https://github.com/nycehs/ehs-data-portal-frontend-temp/tree/gh-pages

But it might have been the github-bot action?
![image](https://user-images.githubusercontent.com/55593359/166260124-f9f2fbfa-0a8b-4a61-ab1d-23544a57ff5a.png)

In short, I am confused. And frustrated! I'm not sure if I broke something that you got working, or if it wasn't quite working properly in the first place.

**Comment by mmontesanonyc** [comment]
Solved? [This commit](https://github.com/nycehs/ehs-data-portal-frontend-temp/commit/41b2714bb502f0e2fc34654dd00ea3f2c3a825ea). 

This `if` statement was in there. I'm not entirely sure what it's testing and I can't find documentation in these two peaceiris repos, either [actions-gh-pages](https://github.com/peaceiris/actions-gh-pages) or [actions-hugo](https://github.com/peaceiris/actions-hugo), but, it wasn't necessary. 

A few other things I found helpful:
- we can remove `/docs` per [this issue](https://github.com/peaceiris/actions-hugo/issues/515)
- [this issue](https://github.com/peaceiris/actions-hugo/issues/578) has some other helpful notes - particularly around `publish_dir` actually being the source directory we're publishing _from_.
- `publish_branch` might not be necessary since the default is `gh-pages` (I think, though I can't find where I read that), but let's leave it in.

Lastly... it's entirely possible that that `if` statement is important, and should actually match the `development` branch (and the statements at the top that trigger the actions) - I don't know. Let's monitor for any unusual behavior. I'll run a few more tests.

**Comment by mmontesanonyc** [comment]
OK... so... not actually successful.

It _did_ solve the problem we had before, in that it ran the deployment part of the hugo-build-to-github-pages action:
![image](https://user-images.githubusercontent.com/55593359/166271008-b097f882-9197-4e0f-a1c8-99f826821c03.png)

But, it still didn't result in a deployment. This is confusing, because it _did_ commit to `gh-pages`:
![image](https://user-images.githubusercontent.com/55593359/166273340-418f0ab3-82b7-4b38-886f-72d9660dbeeb.png)

...but, that branch is showing an error on the `pages-build-deployment` action. 
![image](https://user-images.githubusercontent.com/55593359/166273893-12828054-99f3-4adf-8d38-cb8e20724350.png)

It's failing on "Build page with Jekyll."
![image](https://user-images.githubusercontent.com/55593359/166274579-8eb27537-3305-4371-a818-de6bf1002362.png)

Two thoughts - 
- Does Github Pages use Jekyll to build/deploy non-Jekyll pages? I know that you had added `.nojekyll` to the repo (and I saw that documentation in the peaceiris material). 
- I just noticed that in Settings, gh-pages was set to serve `/docs` instead of `root` so I changed it to root... _BUT_ it was serving the site even though there's no `/docs` in `gh-pages` so that's confusing. 

I'll test having made the latter change.

**Comment by mmontesanonyc** [comment]
OK! That worked. 

![image](https://user-images.githubusercontent.com/55593359/166285605-82699142-f787-4885-9299-fbc0339784ac.png)

I'm seeing the changes to Key Topics pages I hadn't seen before:
![image](https://user-images.githubusercontent.com/55593359/166288914-c8a5e0a1-0e42-499e-bef9-ff3c1044993e.png)

Still slightly confusing is that the `gh-pages` branch isn't showing recent commits, but I think the 'deployment' was committed earlier, it's just that GHP failed to run/serve things properly. It's also showing a failed deployment, but when clicking in to it, the checks have passed. 
![image](https://user-images.githubusercontent.com/55593359/166289142-9ea58b17-e279-4409-b719-947b543637dc.png)

**Comment by mmontesanonyc** [comment]
Related to this - I wonder if we can add a step that runs the search indexing. Or, if our workflow is just to merge, and then re-index on dev, and it's ready for the next merge. But that would leave the search indexing one update behind. Another thing we could do is add to the deployment action a deployment on commit to development, too, so that we could run and commit the search indexing (and the new PagesIndex.json) right to Development, and that would trigger a build/deployment.

**Comment by cgettings** [comment]
I think we can chain basically anything together, since you can run actions based on other actions. I can't remember how the indexing works, but even if it was "build > index + commit > build" I think we could get it to work automatically.

**Comment by cgettings** [comment]
@mmontesanonyc  "its_alive.gif"

**Comment by cgettings** [comment]
- [ ] Switch from push on indexing branch, to merge on development
- [ ] Handle translations?

Re: translations, right now translated content is showing up in results (e.g., asthma matches Spanish). Do we want to handle this differently? Maybe weight non-English content lower?

**Comment by mmontesanonyc** [comment]
Hmmm, interesting point about the translated content - and another wrench in there. Looks like the way that `Gruntfile.js` builds the URLs for the translated content is off, showing this for example:

`https://nycehs.github.io/ehs-data-portal-frontend-temp/data-stories/violence/index.cn` which 404s out. 

![image](https://user-images.githubusercontent.com/55593359/166686896-a5a39a6d-e511-4e10-9e6a-ce1d4ca9cb57.png)
 
I think in general it would be okay to have translated content show up in the results, since we don't have so much that it would make the signal:noise of the results problematic. But, would be good to look at that URL creation.

**Comment by cgettings** [comment]
It does, and it is. Last night I literally 🙌
